packages/snaps-utils/src/json-rpc.ts

```diff
@@ -49,6 +64,36 @@ export function assertIsRpcOrigins(
   );
 }
 
+/**
+ * Check if the given origin is allowed by the given JSON-RPC origins object.
+ *
+ * @param origins - The JSON-RPC origins object.
+ * @param subjectType - The type of the origin.
+ * @param origin - The origin to check.
+ * @returns Whether the origin is allowed.
+ */
+export function isOriginAllowed(
+  origins: RpcOrigins,
+  subjectType: SubjectType,
+  origin: string,
+) {
+  const allowedOrigins =
+    subjectType === SubjectType.Snap ? origins.snaps : origins.dapps;
+
+  // If `allowedOrigins` is true, all origins are allowed.
+  if (allowedOrigins === true) {
+    return true;
+  }
+
+  // If `allowedOrigins` is false or undefined, the origin is not allowed.
+  if (!allowedOrigins) {
+    return false;
+  }
+
+  // Otherwise, check if the origin is in the list of allowed origins.
+  return allowedOrigins.allowedOrigins.includes(origin);
```

Comment:
Is wildcard matching out of scope here? E.g. `*.metamask.io`

We can probably also add that in a future PR.